lib/Conf.ml

```diff
@@ -1864,20 +1864,47 @@ let string_of_user_error = function
   | `Unknown (name, _) -> Format.sprintf "Unknown option %S" name
   | `Bad_value (name, msg) -> Format.sprintf "For option %S: %s" name msg
 
+(** strips [exe] version of the attributes that are not used in [expected]. *)
+let strip_version ~expected ~exe =
+  let open Semver in
+  let Semver.{major; minor; patch; prerelease; build} = exe in
+  let prerelease, build =
```

Comment:
I think that can be simplified to

```
let prelease = if List.is_empty expected.prelease then [] else prerelease in
let build  = if List.is_empty expected.build then [] else build in
```

(or with pattern matching; the point is that the two decisions are independent)